### ISSUE TITLE
Request for Speaker Slides Post-Conference

### ISSUE THREAD

**Issue by k3ii** [Front-End-Coders-Mauritius/frontendmu-astro#44]
Hi there,

I attended the meetup and loved the talks. Would it be possible to access the slides used during the presentations?

Having the slides would help better understand the concepts presented and refresh my memory on the content. It could also benefit those who couldn't attend but are interested in the topics.

**Comment by MrSunshyne** [comment]
@k3ii For now, we don't have the slides stored in a central place. The best option would be to reach out to the speakers individually and ask them. 

We can set up something on the website that would allow linking a file or URL to a session. This requires a little development, but shouldn't be too hard. It's an excellent suggestion, I'll look into implementing this.

**Comment by joomun** [comment]
I agree that having a central place for the presentation slides would be helpful. Here's the plan for how I’m thinking of doing this:

**Plan**
Backend Stuff:

1. I’ll set up some endpoints to handle file uploads and linking URLs to session details. This will let speakers upload their slides directly to our server or link to external URLs.

Frontend Stuff:

1. I’ll create a component for uploading slides, where users can select and upload their files.
2. I’ll also make a component for linking URLs to sessions, where users can enter a session ID and the corresponding URL.

**What I Need from You**
Some minor information:

1. Session Details: Can you share how session details are stored? For example, the structure of the session IDs and any related metadata.
2. File Storage: Are there any specific requirements for where and how the files should be stored? (e.g., local server, cloud storage)
3. Security and Access: Should there be any restrictions on who can upload or link slides? Do we need authentication or authorization checks?

Once you give the go @MrSunshyne and a little bit more info I can give it a go

**Comment by MrSunshyne** [comment]
@joomun I think we can vastly simplify the process with one assumption: the speaker will host their own slides and provide us with a link.

Most slides are urls anyway. Google Slides, Figma, Slidev, github repos, etc..

Advantages: 
- We do not host any files. Lightweight project.
- Avoid copyright issues. Ownership of the presentation stays with speaker.
- no need to implement any uploads or authentication for speakers
- We can link to anything, even a youtube video.

Disadvantages:
- Some speakers might find it difficult to host their presentation if it's a weird format.
- No preview possible. We don't have a thumbnail to display. Only a url.

Todo: 
- This means I only need to add one field in the backend, which is a simple text box that will contain the URL to the slide. 

Plan: 
- When the data is read from the CMS, it will now contain a new field for each session called "deck".
- Deck can be a string or empty.
- If a session has a deck, modify the SessionList component to display a link to the deck.
- Bonus: Display all the links to decks for a single speaker with multiple sessions on their speaker page.

What do you think @joomun  ?

**Comment by joomun** [comment]
Hello

I agree, having the speakers host their own slides and provide us with a link simplifies things a lot. 

And your suggested approach seems more feasible.

This way, speakers can easily share their slides, and we can link to various formats like Google Slides, Figma, Slidev, GitHub repos, YouTube videos, etc.

If you give the go i can try to start work on this feature @MrSunshyne

**Comment by derecklhw** [comment]
@MrSunshyne the slides should be in a specific format ? i.e pdf or pptx ?

**Comment by MrSunshyne** [comment]
@derecklhw Please [check out the plan here](https://github.com/Front-End-Coders-Mauritius/frontend.mu/issues/44#issuecomment-2267143289). tldr; decks are links :D 

@joomun 
I've [updated the ci](https://github.com/Front-End-Coders-Mauritius/frontend.mu/pull/168) to now query the deck field. It will be available [in the meetups-raw.json](https://github.com/Front-End-Coders-Mauritius/frontend.mu/blob/main/packages/frontendmu-data/data/meetups-raw.json) file as either "null" or a string.

if you open the link and CTRL+F "deck" you should see them available already

**Comment by derecklhw** [comment]
@MrSunshyne  check if this is okay i have a [repo with all my slides](https://github.com/derecklhw/speaker_talks)

**Comment by MrSunshyne** [comment]
@derecklhw i've updated your feb 2024 session to include the link to your slide
i've update my may 2024 session about scroll driven animations.

The above decks are now available in the dataset @joomun

**Comment by joomun** [comment]
Well noted thanks @MrSunshyne  will start the dev this week.
